subprojects/core-api/src/main/java/org/gradle/api/file/SourceDirectorySet.java

```diff
@@ -127,4 +128,19 @@
      * @since 4.0
      */
     void setOutputDir(File outputDir);
+
+    /**
+     * Returns the directory property that is, or can be, bound to the task that produces the output.
+     * Use this as part of a classpath or input to another task to ensure that the output is created before it is used.
+     *
+     * Set this property to a output property of a task to connect this source directory set to the output producing (compile) task.
+     *
+     * Note: To define the path of the output folder for a source directory set that is already wired to
+     * a task (e.g. 'java' / `compileJava`) use {@link #setOutputDir(Provider)} or {@link #setOutputDir(File)}.
+     *
+     * @return The output directory property for this set of sources.
+     * @since 6.1
+     */
+    @Incubating
+    DirectoryProperty getOutputDirectoryProperty();
```

Comment:
Is it necessary to have `Property` in the name of the method? I'd love to avoid doing this.

Reply:
It's intentional because there is also `setOuputDir()/getOutputDir()` which are there for a slightly different purpose. But I am open for suggestion to name this differently. (See Javadoc on this method and PR description for why this is.)

Reply:
Maybe we can name it something like `getCompileTaskDestination()` or `getProducingTaskDestination()`...

Reply:
I like using "destination" to differentiate a bit better.

BTW, we also have `destinationDir` vs `destinationDirectory` in the compile task. I thin that's fine, especially if we start deprecating the abbreviated names soon.

Reply:
This has been solved by implementing this suggestion: https://github.com/gradle/gradle/pull/11513#discussion_r352444399